mm2src/coins/eth/web3_transport.rs

```diff
@@ -159,13 +181,43 @@ async fn send_request(
     const REQUEST_TIMEOUT_S: f64 = 60.;
 
     let mut errors = Vec::new();
-    for uri in uris.iter() {
-        let request = to_string(&request);
-        event_handlers.on_outgoing_request(request.as_bytes());
 
-        let mut req = http::Request::new(request.clone().into_bytes());
+    let serialized_request = to_string(&request);
+
+    for node in nodes.iter() {
+        let mut serialized_request = serialized_request.clone();
+        if node.gui_auth {
+            if let Some(generator) = gui_auth_validation_generator.clone() {
+                let signed_message = match EthCoin::generate_gui_auth_signed_validation(generator) {
+                    Ok(t) => t,
+                    Err(e) => {
+                        errors.push(ERRL!(
+                            "GuiAuth signed message generation failed for {:?} node, error: {:?}",
+                            node,
+                            e
+                        ));
+                        continue;
+                    },
+                };
+
+                let auth_request = AuthPayload {
+                    request: &request,
+                    signed_message,
+                };
+
+                serialized_request = to_string(&auth_request);
+            } else {
+                errors.push(ERRL!("GuiAuthValidationGenerator is not provided for {:?} node", node));
+                continue;
+            }
```

Comment:
It's not critical, but this is more readable for me:
```rust
    if node.gui_auth {
            let generator = match gui_auth_validation_generator.clone() {
                Some(gen) => gen,
                None => {
                    errors.push(ERRL!("GuiAuthValidationGenerator is not provided for {:?} node", node));
                    continue;
                },
            };

            let signed_message = match EthCoin::generate_gui_auth_signed_validation(generator) {
                Ok(t) => t,
                Err(e) => {
                    errors.push(ERRL!(
                        "GuiAuth signed message generation failed for {:?} node, error: {:?}",
                        node,
                        e
                    ));
                    continue;
                },
            };

            let auth_request = AuthPayload {
                request: &request,
                signed_message,
            };

            serialized_request = to_string(&auth_request);
        }
```
In that implementation you can easily follow the code branches and can see what will happen in the end (`serialized_request = to_string(&auth_request);`) if all conditions are met.